code/wdb/common.q

```diff
@@ -0,0 +1,528 @@
+                .servers.startupdepcycles[.wdb.tickerplanttypes;.wdb.tpconnsleepintv;.wdb.tpcheckcycles];
+                subscribe[];
+                ];
+        }
+
+/ - if there is data in the wdb directory for the partition, if there is remove it before replay
+/ - is only for wdb processes that are saving data to disk
+clearwdbdata:{[]
+        $[saveenabled and not () ~ key wdbpart:.Q.par[savedir;getpartition[];`];
+                [.lg.o[`deletewdbdata;"removing wdb data (",(delstrg:1_string wdbpart),") prior to log replay"];
+                @[.os.deldir;delstrg;{[e] .lg.e[`deletewdbdata;"Failed to delete existing wdb data.  Error was : ",e];'e }];
+                .lg.o[`deletewdbdata;"finished removing wdb data prior to log replay"];
+                ];
+                .lg.o[`deletewdbdata;"no directory found at ",1_string wdbpart]
+        ];
+        };
+
+/ - function to check that the tickerplant is connected and subscription has been setup
+notpconnected:{[]
+        0 = count select from .sub.SUBSCRIPTIONS where proctype in .wdb.tickerplanttypes, active}
+
+getsortparams:{[]
+        /- get the attributes csv file
+        /-even if running with a sort process should read this file to cope with backups
+        .sort.getsortcsv[.wdb.sortcsv];
+        /- check the sort.csv for parted attributes `p if the writedownmode `partbyattr is selected
+        /- if each table does not have at least one `p attribute the process will exit
+        if[writedownmode~`partbyattr;
+
+                /- check that default table is defined
+                if[not count exec distinct tabname from .sort.params where tabname=`default,att=`p,sort=1b;
+                        .lg.e[`init;"default table not defined in sort.csv with at least one `p attribute and sort=1b"];
+                ];
+                .lg.o[`init;"default table defined in sort.csv and with at least one `p attribute and sort=1b"];
+
+                /- check for `p attributes
+                if[count notparted:distinct .sort.params[`tabname] except distinct exec tabname from .sort.params where att in `p;
+                        .lg.e[`init;"parted attribute p not set at least once in sort.csv for table(s): ", ", " sv string notparted];
+                ];
+                .lg.o[`init;"parted attribute p set at least once for each table in sort.csv"];
+        ];
+        };
+
+\d .
+
+/- get the sort attributes for each table
+.wdb.getsortparams[];
+
+/- Initialise current partiton
+.wdb.currentpartition:.wdb.getpartition[];
+
+/-  adds endofday and endofperiod functions to top level namespace
+endofday: .wdb.endofday;
+endofperiod:{[currp;nextp;data] .lg.o[`endofperiod;"Received endofperiod. currentperiod, nextperiod and data are ",(string currp),", ", (string nextp),", ", .Q.s1 data]};
+
+/- setting the upd and .u.end functions as the .wdb versions
+.u.end:{[pt]
+        .wdb.endofday[.wdb.getpartition[];()!()];
+    }
```

Comment:
to me this stuff all feels like starting to actually initialise the process (there is actually code being run in some of these lines, not just functions/variables being defined etc.), so I think this should be in the `processes/wdb.q` script. Some of it may be duplicated in to the `processes/tailer.q` script as well, but it may or may not need all of it (e.g. not sure if `.wdb.getsortparams` is applicable to tailer) - I think some duplication is OK though. IMO this sort of "library" code shouldn't set/modify any root namespace variables (i.e. there probably shouldn't be a `\d .` in it anywhere)

Reply:
- It was at Gary's request that I moved in lines 515-528 from `processes/wdb.q` & `processes/tailer.q` to minimise duplication: I can revert this so that function calls are ran in these two scripts again, but I'd feel more comfortable if I had consensus from the senior dev team.
- `.wdb.startup[]` throws an error if `.wdb.getsortparams[]` has not been defined: I've traced the chain of dependencies, if an outline of them would be of help?

Reply:
If Gary requested it, that's fine then